### PR TITLE
Define armv7 instruction set for OMZ demos built natively on ARM

### PR DESCRIPTION
**Pull request by alvoron**
When OMZ demos are built natively on ARM platform they could be compiled with `armv6` instruction set (by default Raspbian GCC compiler may use `armv6`). 
OpenVINO Raspbian cross-compilation package is built with `armv7-a` instruction set.
If OMZ demo which was built with `armv6` instruction set links to cross-compiled OpenVINO libraries, then demo is crashed variosly. The following crashed were reported before:
https://github.com/openvinotoolkit/open_model_zoo/issues/2589
https://github.com/openvinotoolkit/open_model_zoo/issues/1852

The fix is to obviously set `armv7-a` instruction set for OMZ demos if demos are built on ARM platform and it's not cross-compilation.

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: demos/CMakeLists.txt [hunk 1]
```diff
@@ -60,6 +60,10 @@ else()
     set(COMPILER_IS_GCC_LIKE FALSE)
 endif()
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm.*|ARM.*)" AND NOT CMAKE_CROSSCOMPILING)
```
Comment (ilya-lavrenov) [inline demos/CMakeLists.txt:63]: CMAKE_SYSTEM_PROCESSOR -> CMAKE_HOST_SYSTEM_PROCESSOR ?

Comment (alvoron) [reply demos/CMakeLists.txt:63]: The [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html) says that `CMAKE_SYSTEM_PROCESSOR` has the same value as the `CMAKE_HOST_SYSTEM_PROCESSOR` when not cross-compiling.
However, probably, `CMAKE_HOST_SYSTEM_PROCESSOR` usage is more robust.

Comment (alvoron) [reply demos/CMakeLists.txt:63]: @ilya-lavrenov do you prefer to use `CMAKE_HOST_SYSTEM_PROCESSOR `?

Comment (ilya-lavrenov) [reply demos/CMakeLists.txt:63]: both are workable, but `CMAKE_HOST_SYSTEM_PROCESSOR` just emphasizes the fact we are building on host :)

**Review by icvgh — APPROVED** [review]

**Review by Wovchena — APPROVED** [review]

**Review by dkurt — APPROVED** [review]

**Review by vladimir-dudnik — APPROVED** [review]

**Pull request merged**